### PR TITLE
[TheWitcher] incorporate crit into roll

### PR DESCRIPTION
**Pull request by Stexinator**
## Roll20 Requests

Comments are very helpful for reviewing the code changes. Please answer the relevant questions below in your comment.

- [x] Does the pull request title have the sheet name(s)? Include each sheet name.
- [ ] Is this a bug fix?
- [x] Does this add functional enhancements (new features or extending existing features) ?
- [ ] Does this add or change functional aesthetics (such as layout or color scheme) ? 
- [ ] If changing or removing attributes, what steps have you taken, if any, to preserve player data ?
- [ ] If this is a new sheet, did you follow [Building Character Sheets standards](https://wiki.roll20.net/Building_Character_Sheets#Roll20_Character_Sheets_Repository) ?

If you do not know English. Please leave a comment in your native language.

### CODE REVIEW

**Comment by Stexinator** [comment]
@Leothedino please review these changes

**Comment by Leothedino** [comment]
@Stexinator HTML clean up is always welcome. Thanks
As for the crit, I see that you encountered the original issue as to why I didn't do this. Roll20 allows exploding dice upward, but cannot deal with exploding downward (on a fumble). I originally had the feature you included now, but became inundated with requests/complaints in my mailbox that it 'the sheet was bugged' or 'only exploding on crit's and not fumbles'. 

If you're going to make this change, I recommend finding a way to make a 'fumble' button in the Rolltemplate of a fumble. That way they can at least roll the fumble exploding dice from the chat window or at the very least rename 'Crit Or Fumble' in the red Quick Roll Buttons box to just 'Fumble'. Personally, as much as I like this, I think it's going to cause major headaches. 

Appreciate the work and for tagging me.

**Comment by Stexinator** [comment]
I changed the template to display as only fumble. The button is already renamed and I changed the text to make it obvious that the crit was already done, but the fumble has to be rolled separatly

File: TheWitcher/witchersheet.html [hunk 1]
```diff
@@ -2680,7 +2680,7 @@
 						<hr style='margin-bottom: 10px; margin-top: 5px;'>
 						<label class='sheet-initiativetitle' data-i18n="initiativ-u" >INITIATIVE</label><button class='sheet-initiative-roll' type='roll' value= '@{whisper}&{template:initiative} {{title=INITIATIVE}} {{name=@{character_name}}} {{roll=[[1d10+@{ref_npc}[STAT]&{tracker}]]}}'></button><br />
 						<label class='sheet-savetitle' data-i18n="stun-death-save-u" >STUN & DEATH SAVE</label><button class='sheet-save-roll' type='roll' value= '@{whisper}&{template:save} {{title=STUN/DEATH SAVE}} {{name=@{character_name}}} {{description= To succeed you need to roll below [[@{stun_npc}]].}} {{roll=[[1d10]]}}'></button><br />
-						<label class='sheet-crittitle' data-i18n="crit-or-fumbl-u" >CRIT OR FUMBLE</label><button class='sheet-critical-roll' type='roll' value= '@{whisper}&{template:crit} {{title=CRIT/FUMBLE}} {{name=@{character_name}}} {{roll=[[1d10]]}}'></button><br />
```
Comment (Stexinator) [inline TheWitcher/witchersheet.html:2683]: see here

File: TheWitcher/witchersheet.html [hunk 2]
```diff
@@ -3371,7 +3371,7 @@
         <tr><th  style='text-transform: uppercase;'>{{title}}</th></tr>
         <tr><td><strong class="template-title">NAME: </strong> {{name}}</td></tr>
         <tr><td><strong class="template-title">SKILL ROLL: </strong> {{roll}}</td></tr>
-        {{#rollWasCrit() roll}}<tr style='text-align: justify; font-size: 12px;'><td><strong class="template-title" style='color: #66c600; text-shadow: 1px 1px #000;'>CRITICAL:</strong> Use the Critical/Fumble button, **add** the result to your total score.</td></tr>{{/rollWasCrit() roll}}
+        {{#rollWasCrit() roll}}<tr style='text-align: justify; font-size: 12px;'><td><strong class="template-title" style='color: #66c600; text-shadow: 1px 1px #000;'>CRITICAL:</strong> Criticals are already calculated into the roll</td></tr>{{/rollWasCrit() roll}}
```
Comment (Stexinator) [inline TheWitcher/witchersheet.html:3381]: and like this

**Comment by Stexinator** [comment]
I check if I can give a one click option to roll the fumble directly from the message. But this will be done in a separate commit

**Comment by Leothedino** [comment]
@Stexinator the button may be renamed, but the translation file is where Roll20 truly takes the name from. It'll need to be adapted/changed in there, too. 

As for the click-on-message, I look forward to seeing what can be done. Tied down and busy these days, so thank you for continuing to support the sheet and for your work. 👍

**Comment by Stexinator** [comment]
@Leothedino yeah translation is an issue. I decided to change the key, so that old translation wont introduce additional confusion

**Comment by Leothedino** [comment]
@Stexinator there really is no need to over complicate it, the .JSON file handles translation automatically. I recommend updating the line for the 'Critical Or Fumble' to whatever it is now. I won't be available to help before this push goes live, but avoid _work-around's_ if you can as it makes it harder for others to contribute/get involved with the sheet.

**Comment by Stexinator** [comment]
@Leothedino renaming the i18n resource is not a workaround. The intention of the button changed so the variables should reflect that, so that it is easier to contribute. I took care of the JSON files as you suggested

**Comment by mperes** [comment]
@Leothedino, @Stexinator this seem good to me. May I go ahead and merge it?

**Comment by Stexinator** [comment]
@mperes I would like to hold it off until next week so I got a chance to test it in custom sheet sandbox

**Comment by Leothedino** [comment]
@Stexinator I had a chance to test your code tonight, it's not working I am afraid. The fumble button is returning `TypeError: Cannot read property 'substring' of undefined` and the 'Crit or Fumble' button appears to be broken. 
Screenshot 2021-06-26 at 02 23 17

**Comment by Stexinator** [comment]
Thats why I said I wanted to test them first :)

**Comment by Stexinator** [comment]
I got some time and run my tests. I corrected the roll template and translation. The only thing missing is to roll directly from the message. I am further looking into that. @Leothedino if you are fine with doing it later I would remove that from this PR, clean it up and call it done

**Comment by Leothedino** [comment]
@Stexinator I won't be fixing this later, or any time soon I am afraid. I have a 'you break it you fix it' policy. Sorry.

 I admit, I am not a fan of this auto-crit change, without a button in the chat box it breaks more than it fixes. You have auto-crits, but manual fumbles. It causes confusion and complaints, the latter that gets emailed to me. I tried to do this before, and people reacted very poorly to it. I suggest not pushing these changes forward until Roll20 update their systems to allow negative and positive exploding dice/rolls.

**Comment by Stexinator** [comment]
@Leothedino I got a tip and got it to work. I would like to give the button a styling which is more in tone with the rolltemplates. Do you have suggestions/mock ups for it?

**Comment by Stexinator** [comment]
@Leothedino I added styling for the button and a safeguard so that when a 10 and then a 1 is rolled it wont show the fumble text. I hope this satisfies your concerns and we can go forward with the merge :)

**Comment by Stexinator** [comment]
@mperes from my point of view this can be merged. :)

**Comment by Leothedino** [comment]
@Stexinator I am really impressed, it's working great. Well done on overcoming the challenges here, it looks and works perfectly. Good job, and from the community and myself, thank you for your contribution!

@mperes all good here. I approve of the merge!

**Comment by Stexinator** [comment]
Happy that you are content with the changes :) I would like to try to do something similar for location and dmg rolls. I make sure to tag you when I create a PR

**Comment by mperes** [comment]
@Stexinator I have merged this one but unfortunatelly I will have to revert it because it contained changes to translations files which are are not allowed anymore, mainly because among other things, it causes troubles with crowdin. Please submit it again without the changes to translations files.

**Comment by Leothedino** [comment]
@Stexinator @mperes I'll take a look tomorrow, I think Stexinator may have manually edited the individual files. I'll see about updating the main translation file. That should be enough.

**Comment by Stexinator** [comment]
new pr at #9175

**Pull request merged**